src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx

```diff
@@ -140,6 +147,41 @@ class RaceTrackerMap extends React.Component<IRaceTrackerMapProps, any> {
     </div>
   );
 
+  private _onClick = (event:any)  => {
+    const feature = event.features && event.features[0];
+    if (feature) {
+        if(feature.layer.id == "point"){
+            this.setState({popup: {
+                lat: event.lngLat[1],
+                lng: event.lngLat[0],
+                description:feature.properties.description
+            } 
+          })
+         
+       }
+    }
+  };
+
+ private  _renderPopup() {
```

Comment:
Referencing my above comment about writing out the HTML, your function would become: 
```
private _renderPopup() {
    const {popup} = this.state;
    return (
      popup && (
        <Popup
          tipSize={5}
          anchor="top"
          longitude={popup.lng}
          latitude={popup.lat}
          closeOnClick={false}
          onClose={() => this.setState({popup: null})}>
          <div>
            <h3>
              Racer ID: {this.state.selectedTrackLeaderId}
            </h3>
            <strong>
              Device Battery State: {this.state.selectedTrackLeaderBatteryState}
            </strong>
          </div>
        </Popup>
      )
    );
  }
```

Reply:
In this example, I assume you've `setState` the data you need, and then reference here.

Reply:
Additionally, `_renderPopup` should become `__renderPopup`, and should also be an arrow function:
`private __renderPopup = () => {};`